app/src/main/java/com/yalta/services/Storage.kt

```diff
@@ -0,0 +1,32 @@
+package com.yalta.services
+
+import androidx.lifecycle.MutableLiveData
+import common.Route
+import javax.inject.Inject
+import javax.inject.Singleton
+
+interface Storage {
+    val routes: MutableLiveData<List<Route>>
+    fun update(updatedRoute: Route)
+    fun clearData()
+}
+
+@Singleton
+class StorageImpl @Inject constructor() : Storage {
+    override val routes = MutableLiveData<List<Route>>(emptyList())
+
+    override fun update(updatedRoute: Route) {
+        if (routes.value != null) {
+            val copy = routes.value?.toMutableList()
+            val item = copy?.find { route -> route.id == updatedRoute.id }
```

Comment:
why do you need a copy here? why can't you search through the original collection?

Reply:
Because `List` allows only to read items
So we can either make it `MutableList` and do whatever we like or do what we want using copies

Reply:
I'd vote for MutableList if it's not too hard?

Reply:
it is completely fucked up